<launch>

	<!-- $ rosrun fiesta nodelet cam_pose_relay-->
	<node pkg="fiesta" type="cam_pose_relay" name="cam_pose_relay" output="screen" />

    <node pkg="sdf_mp_integration" type="fiesta_gpmp2_node" name="fiesta" output="screen"
          required="true">

        <param name="resolution" value="0.05"/>
        <param name="update_esdf_every_n_sec" value="0.1"/>

        <!--Hash Table only-->
        <param name="reserved_size" value="1000000"/>

        <!--Array only-->
        <param name="lx" value="-2.0"/>
        <param name="ly" value="-2.0"/>
        <param name="lz" value="-1"/>
        <param name="rx" value="+2.0"/>
        <param name="ry" value="+2.0"/>
        <param name="rz" value="+2.5"/>

        <!-- raycasting parameters -->
        <param name="min_ray_length" value="0.4"/>
        <param name="max_ray_length" value="4.0"/>
        <!--If hash table is used, make sure the value is 0. In array implementation, 0 means no other thread.-->
        <param name="ray_cast_num_thread" value="1"/>

        <!-- probabilistic grid map -->
        <param name="p_hit" value="0.70"/>
        <param name="p_miss" value="0.15"/>
        <param name="p_min" value="0.12"/>
        <param name="p_max" value="0.97"/>
        <param name="p_occ" value="0.80"/>

        <!-- global / local -->
        <param name="global_map" value="true"/>
        <param name="global_update" value="true"/>
        <param name="global_vis" value="true"/>
        <param name="radius_x" value="2.0"/>
        <param name="radius_y" value="2.0"/>
        <param name="radius_z" value="1.5"/>

        <!--depth_filter -->
        <param name="use_depth_filter" value="true"/>
        <param name="depth_filter_tolerance" value="0.1"/>
        <param name="depth_filter_max_dist" value="4.0"/>
        <param name="depth_filter_min_dist" value="0.1"/>
        <!--unit: pixel-->
        <param name="depth_filter_margin" value="0"/>

        <!-- visulization -->
        <!--0 for no visulize-->
        <param name="visualize_every_n_updates" value="1"/>
        <param name="slice_vis_max_dist" value="2.0"/>
        <!-- relative to the lz if array is used, relative to the origin if hash table is used -->
        <param name="slice_vis_level" value="1"/>
        <param name="vis_lower_bound" value="0"/>
        <param name="vis_upper_bound" value="+4"/>

        <!-- subsribe source -->
        <remap from="~depth" to="/hsrb/head_rgbd_sensor/depth_registered/points"/>
        <remap from="~transform" to="/camera_pose"/>
    </node>

    <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find fiesta)/demo.rviz" />
</launch>
